dlib/test/binary_search_tree.h

```diff
@@ -18,7 +18,6 @@ namespace
 {
 
     using namespace test;
-    using namespace std;
```

Comment:
Don't need to edit any of the stuff in dlib/test.  That is all fine since nothing should be including it.

Reply:
Although I guess it doesn't hurt anything to do it too.

Reply:
I thought, since, I was at it, we might as well do it in the same PR... I can revert those if you want.

Reply:
Na either way is fine.